### ISSUE TITLE
重复的字符似乎特别容易漏识别

### ISSUE THREAD

**Issue by skyoxZ** [kerlomz/captcha_trainer#14]
我是个新手，使用本项目在自己生成的一套验证码上达到了 93% 的识别率。结果似乎表明，当答案中包含连续的相同字符（例如 ABCCDE）时，重复的字符特别容易被漏识别（即识别成 ABCDE）。请问是否有什么建议，可能是哪个地方出了问题？

**Comment by QQ2737499951** [comment]
这个问题我也碰到过，目前无法解决

**Comment by dooozi** [comment]
@skyoxZ 兄弟能帮我看下我的训练结果么？ 看不懂什么意思

INFO:tensorflow:Step: 9200 Time: 13.979 sec/batch, Cost = 0.00025, 160x60-BatchSize: 128
INFO:tensorflow:Step: 9300 Time: 13.981 sec/batch, Cost = 0.00025, 160x60-BatchSize: 128
INFO:tensorflow:Step: 9400 Time: 13.941 sec/batch, Cost = 0.00025, 160x60-BatchSize: 128
INFO:tensorflow:Step: 9500 Time: 14.033 sec/batch, Cost = 0.00024, 160x60-BatchSize: 128
INFO:tensorflow:Step: 9600 Time: 14.009 sec/batch, Cost = 0.00024, 160x60-BatchSize: 128
INFO:tensorflow:Step: 9700 Time: 13.956 sec/batch, Cost = 0.00024, 160x60-BatchSize: 128
INFO:tensorflow:Step: 9800 Time: 14.029 sec/batch, Cost = 0.00023, 160x60-BatchSize: 128
INFO:tensorflow:Step: 9900 Time: 14.055 sec/batch, Cost = 0.00023, 160x60-BatchSize: 128
INFO:tensorflow:Step: 10000 Time: 14.234 sec/batch, Cost = 0.00023, 160x60-BatchSize: 128
INFO:tensorflow:0 7 7 [13, 11, 26, 30, 13, 18, 11] [13, 11, 26, 30, 13, 18, 11] --> ['c', 'a', 'p', 't', 'c', 'h', 'a'] ['c', 'a', 'p', 't', 'c', 'h', 'a']
INFO:tensorflow:1 7 7 [13, 11, 26, 30, 13, 18, 11] [13, 11, 26, 30, 13, 18, 11] --> ['c', 'a', 'p', 't', 'c', 'h', 'a'] ['c', 'a', 'p', 't', 'c', 'h', 'a']
INFO:tensorflow:2 7 7 [13, 11, 26, 30, 13, 18, 11] [13, 11, 26, 30, 13, 18, 11] --> ['c', 'a', 'p', 't', 'c', 'h', 'a'] ['c', 'a', 'p', 't', 'c', 'h', 'a']
INFO:tensorflow:3 7 7 [13, 11, 26, 30, 13, 18, 11] [13, 11, 26, 30, 13, 18, 11] --> ['c', 'a', 'p', 't', 'c', 'h', 'a'] ['c', 'a', 'p', 't', 'c', 'h', 'a']
INFO:tensorflow:4 7 7 [13, 11, 26, 30, 13, 18, 11] [13, 11, 26, 30, 13, 18, 11] --> ['c', 'a', 'p', 't', 'c', 'h', 'a'] ['c', 'a', 'p', 't', 'c', 'h', 'a']
ERROR:tensorflow:[]
INFO:tensorflow:Epoch: 1358, Step: 10000, Accuracy = 1.0000, Cost = 0.00023, Time = 15.367 sec/batch, LearningRate: 0.0009800000116229057
INFO:tensorflow:CNN Output: (?, ?, 256)
INFO:tensorflow:./model/myfirsttest.model-10000
INFO:tensorflow:Restoring parameters from ./model/myfirsttest.model-10000
INFO:tensorflow:Froze 40 variables.
Converted 40 variables to const ops.
INFO:tensorflow:Total Time: 33.93611812591553 sec.
INFO:tensorflow:Training completed.

**Comment by jewelcai** [comment]
感觉ctc的问题  貌似难解决

**Comment by lengthmin** [comment]
期待一场讨论

**Comment by xinyvz** [comment]
因为ctc编码两个相同字符之间必须要插入空，如果池化多了，输入ctc的timestep的长度太短，可能没有足够timestep来插入空，两个相同的m就会合并。可以试试增加输入的长度，或是减少池化层。

**Comment by Funny-dot** [comment]
大佬们，你们的config_demo.yaml文件在哪下载的呀，我咋没看见

**Comment by 634739726** [comment]
> 因为ctc编码两个相同字符之间必须要插入空，如果池化多了，输入ctc的timestep的长度太短，可能没有足够timestep来插入空，两个相同的m就会合并。可以试试增加输入的长度，或是减少池化层。

请问下这个插入的空字符在哪里写呢

**Comment by skyoxZ** [comment]
解决了，如上面几位所说的，不要用 CTC，改用 CrossEntropy 就可以啦
